control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -4234,12 +4244,14 @@ func (r *HostedControlPlaneReconciler) reconcileRouter(ctx context.Context, hcp
 			return fmt.Errorf("failed to reconcile router deployment: %w", err)
 		}
 
-		pdb := manifests.RouterPodDisruptionBudget(hcp.Namespace)
-		if _, err := createOrUpdate(ctx, r.Client, pdb, func() error {
-			ingress.ReconcileRouterPodDisruptionBudget(pdb, hcp.Spec.ControllerAvailabilityPolicy, config.OwnerRefFrom(hcp))
-			return nil
-		}); err != nil {
-			return fmt.Errorf("failed to reconcile router pod disruption budget: %w", err)
+		if hcp.Spec.ControllerAvailabilityPolicy == hyperv1.HighlyAvailable {
```

Comment:
All the Reconcile*PodDisruptionBudget funcs already receive `hcp.Spec.ControllerAvailabilityPolicy` as parameter and call `util.ReconcilePodDisruptionBudget` underneath, so it doesn't make sense to wrap them within `hcp.Spec.ControllerAvailabilityPolicy`. The implementation of util.ReconcilePodDisruptionBudget dictates the output and it can be unit tested

Reply:
Ok, let me change this a little bit. I will remove the availability from the `util.ReconcilePodDisruptionBudget` function to return always the same configuration.